include/ppx/metrics.h

```diff
@@ -122,48 +144,74 @@ class MetricGauge final
         double value;
     };
 
+    struct Stats
+    {
+        // Basic - updated every entry.
+        double min       = std::numeric_limits<double>::max();
+        double max       = std::numeric_limits<double>::min();
+        double average   = 0.0;
+        double timeRatio = 0.0;
+
+        // Complex - computed on request.
+        double median            = 0.0;
+        double standardDeviation = 0.0;
+        double percentile01      = 0.0;
+        double percentile05      = 0.0;
+        double percentile10      = 0.0;
+        double percentile90      = 0.0;
+        double percentile95      = 0.0;
+        double percentile99      = 0.0;
+    };
+
 private:
     MetricGauge(const MetricMetadata& metadata)
-        : mMetadata(metadata) {}
+        : mMetadata(metadata)
+    {
+        PPX_ASSERT_MSG(mMetadata.type == MetricType::kGauge, "Gauge must be instantiated with gauge-type metadata!");
+    }
     METRICS_NO_COPY(MetricGauge)
 
-    void UpdateBasicStatistics(double seconds, double value);
+    Stats ComputeStats() const;
 
 private:
     MetricMetadata               mMetadata;
     std::vector<TimeSeriesEntry> mTimeSeries;
-    GaugeBasicStatistics         mBasicStatistics;
+    Stats                        mBasicStats;
     double                       mAccumulatedValue = 0.0;
 };
 
 ////////////////////////////////////////////////////////////////////////////////
 
 // A counter metric represents a value that only goes up, e.g. the number
 // of stutters or pipeline cache misses.
-class MetricCounter final
+class MetricCounter final : public Metric
 {
     friend class Run;
 
 public:
-    void     Increment(uint64_t add);
-    uint64_t Get() const;
+    ~MetricCounter() override {}
 
-    // Information from the metadata.
-    std::string GetName() const
-    {
-        return mMetadata.name;
-    };
+    bool RecordEntry(const MetricData& data) override;
 
-    nlohmann::json Export() const;
+    // Gets the count of recorded entries.
+    size_t GetEntryCount() const override;
```

Comment:
This might not be useful. I added it for the tests, but ultimately just relied on the Export function. Let me know if you agree!